### PR TITLE
Add IP range

### PR DESCRIPTION
**Pull request by ecbaldwin**
This is a WIP. Right now, I'm just looking to get feedback on the
proposed interface for ranges. Once the interface is agreed upon, I'll
write tests and then implement it.

### CODE REVIEW

File: iprange.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+// IPRange contains a single contiguous range of IP addresses. A valid range
+// cannot be empty, it must have an least one IP address. For this reason, they
+// should be created with the CreateRange method below.
+type IPRange struct {
+}
+
+// CreateRange creates a new range given the two IP addresses passed in. The
+// first address must be less than or equal to the last. The two addresses must
+// be from the IP family (i.e. ipv4 or ipv6)
+func CreateRange(first, last net.IP) (s *IPRange) {
```
Comment (HenryGessau) [inline iprange.go]: Should also return error?

File: iprange.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+// IPRange contains a single contiguous range of IP addresses. A valid range
+// cannot be empty, it must have an least one IP address. For this reason, they
+// should be created with the CreateRange method below.
+type IPRange struct {
+}
+
+// CreateRange creates a new range given the two IP addresses passed in. The
+// first address must be less than or equal to the last. The two addresses must
+// be from the IP family (i.e. ipv4 or ipv6)
```
Comment (ecbaldwin) [inline iprange.go]: "from the same IP family"

File: iprange.go [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+// First returns the first IP in the range.
+func (s *IPRange) First() net.IP {
+}
+
+// First returns the last IP in the range.
+func (s *IPRange) Last() net.IP {
+}
+
+// ToSet converts the range into a IPSet
```
Comment (ecbaldwin) [inline iprange.go]: into an IPSet

**Comment by smikulcik** [comment]
@ecbaldwin  Sorry, I started working on a project where I needed to efficiently compute differences in ranges of IPs, so I happened to reimplement what you have here with a focus on set differences. See #14

**Comment by HenryGessau** [comment]
Superseded by #14

**Pull request closed without merging**